sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -221,10 +221,17 @@ def create_job_resources(options,  # type: PipelineOptions
                 setup_options.requirements_file, REQUIREMENTS_FILE))
         # Populate cache with packages from the requirement file option and
         # stage the files in the cache.
-        (
-            populate_requirements_cache if populate_requirements_cache else
-            Stager._populate_requirements_cache)(
-                setup_options.requirements_file, requirements_cache_path)
+        if setup_options.view_as(WorkerOptions).sdk_container_image is None:
+          # downloads the binary distributions
+          (
+              populate_requirements_cache if populate_requirements_cache else
+              Stager._populate_requirements_cache_with_whl)(
```

Comment:
I would keep one method: _populate_requirements_cache, and add add a flag such as `prefer_bdists=True/False` (or `fetch_binary` to match other flags) to avoid code duplication

Reply:
Added.